samples/net/secure_mqtt_sensor_actuator/Kconfig

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2024 Analog Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+mainmenu "Secure MQTT Sensor Actuator Sample Application"
+
+config MQTT_BROKER_IPV4_ADDR
+	string "IPv4 Address of MQTT Broker"
+	default "91.121.93.94"
+	help
+	  IP address of the MQTT broker the application should connect to.
+	  This is the IP of test.mosquitto.org
+
+config MQTT_BROKER_HOSTNAME
+	string "TLS Server Name Identification"
+	default "test.mosquitto.org"
+	help
+	  This value must match the Common Name (CN) field in
+	  the server's CA certificate.
+
+config MQTT_BROKER_PORT
+	int "MQTT Broker Connection Port"
+	default 8883
+	help
+	  Port through which the application should connect to the MQTT broker.
+	  Secure MQTT uses port 8883.
+
+config MQTT_PUB_TOPIC
+	string "The MQTT topic the application should publish data to"
+	default "zephyr_sample/sensor"
+
+config MQTT_SUB_TOPIC_CMD
+	string "The MQTT topic the application will receive commands on"
+	default "zephyr_sample/command"
+
+config MQTT_PUBLISH_INTERVAL
+	int "Interval between MQTT publishes (in seconds)"
+	default 3
+	help
+	  This config determines the frequency at which the timer interrupt
+	  occurs, which queues an MQTT publish.
+
+config MQTT_PUBLISH_QOS
+	int "Quality of Service used for MQTT publish"
+	default 1
+	help
+	  MQTT_QOS_0_AT_MOST_ONCE = 0
+	  MQTT_QOS_1_AT_LEAST_ONCE = 1
+	  MQTT_QOS_2_EXACTLY_ONCE = 2
```

Comment:
The benefit of the current implementation is that the Kconfig symbol for QoS is an int and its value translates directly over to the enum type for QoS in the MQTT lib. This allows the symbol to be used directly in the code to set the MQTT QoS, which makes things nice and simple.

Is using `choice` a must-have?

Reply:
It would be better if we do not force the user to know what each value means.
You can build you code such a way that the mapping from this config symbol to int value is done at built time.
```
IS_ENABLED(CONFIG_NET_SAMPLE_MQTT_QOS_AT_MOST_ONCE) ? 0 : 
(IS_ENABLED(CONFIG_NET_SAMPLE_MQTT_QOS_AT_LEAST_ONCE) ? 1 : 2)
```